docs/styleguide.md

```diff
@@ -0,0 +1,12 @@
+# styleguide
+
+The sidetree specification relies on JSON and the current node implementations rely on HTTP.
+
+Unless otherwise noted, the following styleguides should be used.
+
+- [Microsoft API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md)
```

Comment:
Those guides are not the same, I just added the 2 most popular ones I know.

Reply:
@kdenhartog @OR13 @csuwildcat Any further action suggested on this?

Reply:
My preference would be that we pick the Microsoft one since it looked the most comprehensive and having only one would prevent us from running into minor inconsistencies between them.